python/iceberg/api/types/conversions.py

```diff
@@ -42,34 +42,34 @@ class Conversions(object):
     to_byte_buff_mapping = {TypeID.BOOLEAN: lambda type_id, value: struct.pack("<h", 1 if value else 0),
                             TypeID.INTEGER: lambda type_id, value: struct.pack("<i", value),
                             TypeID.DATE: lambda type_id, value: struct.pack("<i", value),
-                            TypeID.LONG: lambda type_id, value: struct.pack("<l", value),
-                            TypeID.TIME: lambda type_id, value: struct.pack("<l", value),
-                            TypeID.TIMESTAMP: lambda type_id, value: struct.pack("<l", value),
+                            TypeID.LONG: lambda type_id, value: struct.pack("<q", value),
```

Comment:
why the change here?

Reply:
@rymurr  Here, `<q` is for `long long` with 8 bytes and `<i` is for `integer` with 4 bytes.
I think here we need 8 bytes to match Java. Also `from_byte_buff_mapping` uses `<q`.

Reply:
ahhh, got it. I thought `long long` meant `double long` not `long` and `l` meant `long`. How bizarre of python...